### PR TITLE
SJRK-246: Update intro page content for "read-only" site

### PR DESCRIPTION
**Pull request by BlueSlug**
This consists of a major redesign of the site while still keeping the work under the "learningReflections" theme.

Three new pages have been added:
- introduction.html (the "home" page)
- workshops.html
- resources.html

There is also a link to the staging version of the Edit page in order to allow people to try out the publishing features of the tool without interfering with the stories we've already gathered on the production version of the site.

There are still a few styling issues that need to be dealt with at the time of filling:
- the top nav menu does not work well at medium widths
- text spacing and sizes need to be refined to more closely match the design

@cindyli to provide code review with CSS assistance from @greatislander

### CODE REVIEW

**Comment by cindyli** [comment]
With UIO, if you set text size to 2, line spacing to 2, font to "Comic Sans", there's an overlapping on the header:
Screen Shot 2019-05-24 at 10 16 41 AM

In addition to preferences above, setting a high contrast theme, there is more overlapping on the text "THE STORYTELLING PROJECT":
Screen Shot 2019-05-24 at 10 17 53 AM

The same issue can be produced with other font styles but "Comic Sans" is the most obvious one.

**Comment by cindyli** [comment]
With UIO, setting text size to 2, 3 logos at the footer stretches out of the screen:
Screen Shot 2019-05-24 at 10 23 04 AM

**Comment by cindyli** [comment]
With UIO, turning on "table of content", on "Resources", "Browse Stories", "View a story" pages, the title of "table of content" shows up without having actual table of content:
Screen Shot 2019-05-24 at 10 24 44 AM

**Comment by cindyli** [comment]
On the "workshops" page, with UIO, set font to "Comic Sans", turn on "table of content", there's overlapping between lines of table of content:
Screen Shot 2019-05-24 at 10 30 52 AM

**Comment by cindyli** [comment]
Open debugger -> "console" tab, reload any of these pages: workshops, resources, browse stories, view a story, this error will be thrown: `TypeError: response is undefined`:
Screen Shot 2019-05-24 at 10 33 15 AM

File: ui/src/learningReflections/js/learningReflections.js [hunk 1]
```diff
@@ -14,9 +14,25 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
...
     fluid.defaults("sjrk.storyTelling.learningReflections", {
         gradeNames: ["sjrk.storyTelling.page"],
+        pageSetup: {
+            browseLinkUrl: "storyBrowse.html",
+            buildLinkUrl: "storyEdit.html"
+        },
         components: {
-            // masthead/banner section
-            learningReflectionsMasthead: {
+            menu: {
```
Comment (cindyli) [inline ui/src/learningReflections/js/learningReflections.js]: Line 22-34 can be simplified by replacing it with a distributeOptions block.

**Comment by BlueSlug** [comment]
> With UIO, setting text size to 2, 3 logos at the footer stretches out of the screen:
> Screen Shot 2019-05-24 at 10 23 04 AM

This is a known issue, documented in [SJRK-206](https://issues.fluidproject.org/browse/SJRK-206)

**Comment by BlueSlug** [comment]
> With UIO, turning on "table of content", on "Resources", "Browse Stories", "View a story" pages, the title of "table of content" shows up without having actual table of content:
> Screen Shot 2019-05-24 at 10 24 44 AM

This is a known issue, documented in [SJRK-206](https://issues.fluidproject.org/browse/SJRK-206)

File: ui/src/learningReflections/js/learningReflections.js [hunk 2]
```diff
@@ -14,31 +14,32 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
     // Provides the Learning Reflections framing to the Storytelling Tool
     fluid.defaults("sjrk.storyTelling.learningReflections", {
         gradeNames: ["sjrk.storyTelling.page"],
+        distributeOptions: [{
+            target: "{that > menu > templateManager}.options.templateConfig.templatePath",
+            record: "%resourcePrefix/src/learningReflections/templates/learningReflections-menu.handlebars"
+        },
+        {
```
Comment (cindyli) [inline ui/src/learningReflections/js/learningReflections.js:21]: Two distributeOptions blocks at Line 21-28 are for configuring the same option of a same component, I feel combining them into one makes it easier to read:
```
{
    record: {
        messagesPath: "%resourcePrefix/src/learningReflections/messages/learningReflectionMessages.json",
        templatePath: "%resourcePrefix/src/learningReflections/templates/learningReflections-masthead.handlebars"
    }
    target: "{that > masthead > templateManager}.options.templateConfig",
}
```

The same comment applies to line 29-36 and other places. Thanks.

File: ui/src/learningReflections/js/learningReflections.js [hunk 3]
```diff
@@ -132,31 +89,66 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
...
         },
+        {
+            target: "{that > storyPreviewer > templateManager}.options.templateConfig.templatePath",
+            record: "%resourcePrefix/src/learningReflections/templates/learningReflections-storyViewer.handlebars"
+        }]
+    });
+
+    fluid.defaults("sjrk.storyTelling.learningReflections.introduction", {
+        gradeNames: ["sjrk.storyTelling.learningReflections", "sjrk.storyTelling.page"],
+        distributeOptions: [{
```
Comment (cindyli) [inline ui/src/learningReflections/js/learningReflections.js:102]: The comment above applies to here too.

File: ui/src/learningReflections/js/learningReflections.js [hunk 4]
```diff
@@ -132,31 +89,66 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
...
-                    }
-                }
+                container: ".sjrkc-st-introduction"
+            }
+        }
+    });
+
+    fluid.defaults("sjrk.storyTelling.learningReflections.workshops", {
+        gradeNames: ["sjrk.storyTelling.learningReflections", "sjrk.storyTelling.page"],
+        distributeOptions: [{
```
Comment (cindyli) [inline ui/src/learningReflections/js/learningReflections.js:119]: same here.

File: ui/src/learningReflections/js/learningReflections.js [hunk 5]
```diff
@@ -132,31 +89,66 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
...
+            workshops: {
+                type: "sjrk.storyTelling.ui",
+                container: ".sjrkc-st-workshops"
+            }
+        }
+    });
+
+    fluid.defaults("sjrk.storyTelling.learningReflections.resources", {
+        gradeNames: ["sjrk.storyTelling.learningReflections", "sjrk.storyTelling.page"],
+        distributeOptions: [{
```
Comment (cindyli) [inline ui/src/learningReflections/js/learningReflections.js:136]: same here too.

**Comment by cindyli** [comment]
"Explore Tool" link leads to [the story builder page on the staging site](http://staging-stories.floeproject.org/storyEdit.html). Is this what to be expected on the production site?

**Comment by BlueSlug** [comment]
> "Explore Tool" link leads to [the story builder page on the staging site](http://staging-stories.floeproject.org/storyEdit.html). Is this what to be expected on the production site?

Indeed, the idea is that after this gets merged into stories-floe-dev, it will be merged into stories-floe-production, and then I'll re-enable editing and make another PR to the dev branch (or you could edit the flags directly, I suppose). That way users can test out the tool in the staging site (as noted above), where they'll be notified that anything they create will be deleted within 24 hours.

**Comment by cindyli** [comment]
Why does turning on UIO "table of content" on any of the new pages not populating any table of content, except to display the title? It looks all these pages have html header elements.

**Comment by cindyli** [comment]
> Why does turning on UIO "table of content" on any of the new pages not populating any table of content, except to display the title? It looks all these pages have html header elements.

Let's address this in a separate JIRA if it is an issue.

**Comment by cindyli** [comment]
Merged at a32cb6db323d86fde8ef545c2b884d7e317bddea

**Comment by BlueSlug** [comment]
> Why does turning on UIO "table of content" on any of the new pages not populating any table of content, except to display the title? It looks all these pages have html header elements.

This is the same as the other issue you mentioned earlier, another instance of the problem described in [SJRK-206](https://issues.fluidproject.org/browse/SJRK-206). It's because UIO is building its table of contents before the content is loaded into the DOM by the templateManagers

**Comment by BlueSlug** [comment]
> > Why does turning on UIO "table of content" on any of the new pages not populating any table of content, except to display the title? It looks all these pages have html header elements.
> 
> This is the same as the other issue you mentioned earlier, another instance of the problem described in [SJRK-206](https://issues.fluidproject.org/browse/SJRK-206). It's because UIO is building its table of contents before the content is loaded into the DOM by the templateManagers

To be clear: the table of contents works the first time, if the preference isn't already set to "on". If it's already on at page load, then it acts faster and the content is loaded after the TOC is compiled.

**Pull request merged**